plugins/plugin-languageserver/che-plugin-languageserver-ide/src/main/java/org/eclipse/che/plugin/languageserver/ide/LanguageServerFileTypeRegister.java

```diff
@@ -90,40 +84,43 @@ public void start(final Callback<WsAgentComponent, Exception> callback) {
         registeredLanguages.then(new Operation<List<LanguageDescription>>() {
             @Override
             public void apply(List<LanguageDescription> langs) throws OperationException {
+                LOGGER.info("registering language descriptions");
                 if (!langs.isEmpty()) {
                     JsArrayString contentTypes = JsArrayString.createArray().cast();
                     for (LanguageDescription lang : langs) {
-                        String primaryExtension = lang.getFileExtensions().get(0);
-                        for (String ext : lang.getFileExtensions()) {
-                            final FileType fileType = new FileType(resources.file(), ext);
-                            fileTypeRegistry.registerFileType(fileType);
-                            editorRegistry.registerDefaultEditor(fileType, editorProvider);
-                            ext2langId.put(ext, lang.getLanguageId());
+                        if (lang.getFileExtensions() != null) {
```

Comment:
That sounds good, but it's not easily possible since LanguageDescription has a setter for the file extensions. So I can't really guarantee that it's not null.